### ISSUE TITLE
[BUG] CPU usage increase in jupyter notebook when re-running py3Dmol's animate

### ISSUE THREAD

**Issue by stgeo** [3dmol/3Dmol.js#622]
**Describe the bug**
When creating an animation in a jupyter notebook and then re-running the respective cell, CPU usage climbs with each execution, both in Chromium and Firefox, with the latter outright crashing after a while. This might be related to #520.

**To Reproduce**
In a jupyter notebook, execute the following:
<details>
<summary>Long code snippet</summary>

```
import py3Dmol

moltrj = """
     RDKit          3D

  3  2  0  0  0  0  0  0  0  0999 V2000
    0.0000    0.0000    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
    1.0000    0.0000    0.0000 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.0000    0.0000    0.0000 H   0  0  0  0  0  0  0  0  0  0  0  0
  2  1  1  0
  3  1  1  0
M  END
$$$$

     RDKit          3D

  3  2  0  0  0  0  0  0  0  0999 V2000
    0.0000   -0.0240    0.0137 O   0  0  0  0  0  0  0  0  0  0  0  0
    1.0000    0.1901   -0.1089 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.0000    0.1901   -0.1089 H   0  0  0  0  0  0  0  0  0  0  0  0
  2  1  1  0
  3  1  1  0
M  END
$$$$

     RDKit          3D

  3  2  0  0  0  0  0  0  0  0999 V2000
    0.0000   -0.0388    0.0222 O   0  0  0  0  0  0  0  0  0  0  0  0
    1.0000    0.3076   -0.1762 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.0000    0.3076   -0.1762 H   0  0  0  0  0  0  0  0  0  0  0  0
  2  1  1  0
  3  1  1  0
M  END
$$$$

     RDKit          3D

  3  2  0  0  0  0  0  0  0  0999 V2000
    0.0000   -0.0388    0.0222 O   0  0  0  0  0  0  0  0  0  0  0  0
    1.0000    0.3076   -0.1762 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.0000    0.3076   -0.1762 H   0  0  0  0  0  0  0  0  0  0  0  0
  2  1  1  0
  3  1  1  0
M  END
$$$$

     RDKit          3D

  3  2  0  0  0  0  0  0  0  0999 V2000
    0.0000   -0.0240    0.0137 O   0  0  0  0  0  0  0  0  0  0  0  0
    1.0000    0.1901   -0.1089 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.0000    0.1901   -0.1089 H   0  0  0  0  0  0  0  0  0  0  0  0
  2  1  1  0
  3  1  1  0
M  END
$$$$

     RDKit          3D

  3  2  0  0  0  0  0  0  0  0999 V2000
    0.0000   -0.0000    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
    1.0000    0.0000   -0.0000 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.0000    0.0000   -0.0000 H   0  0  0  0  0  0  0  0  0  0  0  0
  2  1  1  0
  3  1  1  0
M  END
$$$$

     RDKit          3D

  3  2  0  0  0  0  0  0  0  0999 V2000
    0.0000    0.0240   -0.0137 O   0  0  0  0  0  0  0  0  0  0  0  0
    1.0000   -0.1901    0.1089 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.0000   -0.1901    0.1089 H   0  0  0  0  0  0  0  0  0  0  0  0
  2  1  1  0
  3  1  1  0
M  END
$$$$

     RDKit          3D

  3  2  0  0  0  0  0  0  0  0999 V2000
    0.0000    0.0388   -0.0222 O   0  0  0  0  0  0  0  0  0  0  0  0
    1.0000   -0.3076    0.1762 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.0000   -0.3076    0.1762 H   0  0  0  0  0  0  0  0  0  0  0  0
  2  1  1  0
  3  1  1  0
M  END
$$$$

     RDKit          3D

  3  2  0  0  0  0  0  0  0  0999 V2000
    0.0000    0.0388   -0.0222 O   0  0  0  0  0  0  0  0  0  0  0  0
    1.0000   -0.3076    0.1762 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.0000   -0.3076    0.1762 H   0  0  0  0  0  0  0  0  0  0  0  0
  2  1  1  0
  3  1  1  0
M  END
$$$$

     RDKit          3D

  3  2  0  0  0  0  0  0  0  0999 V2000
    0.0000    0.0240   -0.0137 O   0  0  0  0  0  0  0  0  0  0  0  0
    1.0000   -0.1901    0.1089 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.0000   -0.1901    0.1089 H   0  0  0  0  0  0  0  0  0  0  0  0
  2  1  1  0
  3  1  1  0
M  END
$$$$
"""

viewer = py3Dmol.view(width=300, height=300)
viewer.addModelsAsFrames(moltrj, 'sdf')
viewer.setStyle({'stick': {}, 'sphere': {'scale': 0.3}})
viewer.animate({'loop': "forward"})
viewer.zoomTo()
viewer.show()
```

</details>

Run the cell multiple times and monitor CPU usage. In chromium I get 10% for one run, 26% for 5, and 35% for 10 runs.

**Expected behavior**
The animation in the no longer visible viewer should be stopped and it should be destroyed. It seems this is not happening.

**Desktop**
 - OS: Ubuntu 22.04
 - Browser Firefox 105.0.2, Chromium 106.0.5249.9, both installed via snap
 - py3dmol 1.8.0
 - jupyter 1.0.0
 - jupyter-console 6.4.4
 - jupyter-core 4.11.1
 - jupyter_client 7.3.5

**Additional context**
The console output of the browser produces many of the following lines at every execution:
```
[1026375:1026375:1005/121306.369914:ERROR:gl_utils.cc(315)] [.WebGL-0x558f51ced440] GL_INVALID_FRAMEBUFFER_OPERATION: Framebuffer is incomplete: Attachment has zero size.
[1026375:1026375:1005/121306.371367:ERROR:gl_utils.cc(315)] [.WebGL-0x558f51ced440] GL_INVALID_FRAMEBUFFER_OPERATION: Draw framebuffer is incomplete
[1026375:1026375:1005/121306.372229:ERROR:gl_utils.cc(315)] [.WebGL-0x558f51ced440] GL_INVALID_FRAMEBUFFER_OPERATION: Draw framebuffer is incomplete
```

**Comment by dkoes** [comment]
This does look like #520.  In the fix there I stopped the animation when a canvas was removed from the DOM AND the WebGL context was lost.  The reason you are seeing additional memory usage is Chrome is not deactivating the WebGL context (at least not until you've created a bunch of them).  Arguably, halting animation when a canvas is removed from the DOM is not unreasonable behavior so I'll remove the check of the lost context and see what happens.
